src/Weeder/Main.hs

```diff
@@ -104,6 +104,22 @@ parseCLIArguments = do
     pure CLIArguments{..}
 
 
+data Weed = Weed
+  { weedPath :: FilePath
+  , weedLoc :: RealSrcLoc
+  , weedDeclaration :: Declaration
+  , weedPrettyPrintedType :: Maybe String
+  }
+
+
+instance Show Weed where
+  show Weed{..} =
+    weedPath <> ":" <> show ( srcLocLine weedLoc ) <> ": "
+      <> case weedPrettyPrintedType of
+        Nothing -> occNameString ( declOccName weedDeclaration )
+        Just t -> "(Instance) :: " <> t
```

Comment:
I generally don't like using `Show` to produce strings that don't look like Haskell code. I think we'd be fine with just:

```suggestion
formatWeed :: Weed -> String
formaWeed Weed{..} =
  weedPath <> ":" <> show ( srcLocLine weedLoc ) <> ": "
    <> case weedPrettyPrintedType of
      Nothing -> occNameString ( declOccName weedDeclaration )
      Just t -> "(Instance) :: " <> t
```